### PR TITLE
[java-gen] feat: Maven Plugin + Integration test infrastructure

### PR DESCRIPTION
**Pull request by manusa**
## Description
- Spiked a Maven Plugin to invoke the generator.
- Created an integration test infrastructure based on the plugin
  - TODO: Test generated sources compared to our actual extensions (the end goal would be to be able to replace our current code generation with this --> Configuration options should be added for package, additionalProperties generation, etc.)
  - TODO: Create a proper verification infrastructure

Other comments:
- Current implementation is slow. We can probably improve the performance by running the generation in multiple threads for each individual CRD.
- CRD Generator only admits passing in a File. We could support reading complete directories (makes a lot of sense for the Maven Plugin workflow)

/cc @andreaTP 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by andreaTP** [comment]
This is awesome, an extremely appreciated and amazing addition! 🙏 

Do you think that we should, eventually, leverage this infrastructure instead of using the `maven-invoker` for integration tests? ( [ref](https://github.com/fabric8io/kubernetes-client/pull/3812/files#diff-3da10905fd06ededa34a8e3bb75f061210de4f1e799ddb2d55f87511965382e4R139) )

Regarding:

> Current implementation is slow. We can probably improve the performance by running the generation in multiple threads for each individual CRD.

I think this is a great starting point and we can get there in the future.

> CRD Generator only admits passing in a File. We could support reading complete directories (makes a lot of sense for the Maven Plugin workflow)

Agreed, maybe we should scan specific directories at some point e.g. `kubernetes`

**Review by rohanKanojia — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

File: java-generator/core/pom.xml [hunk 1]
```diff
@@ -99,8 +99,6 @@
     <dependency>
       <groupId>com.approvaltests</groupId>
       <artifactId>approvaltests</artifactId>
-      <version>13.3.0</version>
-      <scope>test</scope>
```
Comment (andreaTP) [inline java-generator/core/pom.xml:103]: this way we are adding `approvaltests` to the transitive dependencies, is it intended?

File: java-generator/it/src/it/cert-manager/verify.groovy [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import static org.junit.jupiter.api.Assertions.assertEquals
+
+// TODO: This should be modified to test against the current extension sources
+assertEquals(
```
Comment (andreaTP) [inline java-generator/it/src/it/cert-manager/verify.groovy:19]: should we use `approvaltests` here?

**Review by andreaTP — COMMENTED** [review]
A couple of questions, but the bulk of it is great work! 👍

File: java-generator/it/src/it/cert-manager/verify.groovy (hunk 2 above)
Comment (manusa) [reply java-generator/it/src/it/cert-manager/verify.groovy:19]: I tried to, but it didn't work (Approval tests relies on a JUnit execution environment).

Anyway, considering the way we are using Approvaltests, this is practically the same (minus the dependency)

File: java-generator/core/pom.xml (hunk 1 above)
Comment (andreaTP) [reply java-generator/core/pom.xml:103]: 👍

File: java-generator/it/src/it/cert-manager/verify.groovy (hunk 2 above)
Comment (andreaTP) [reply java-generator/it/src/it/cert-manager/verify.groovy:19]: There is a port for Groovy:
https://github.com/rzabini/ApprovalTests.Spock

What I'm enjoining from the Approval tests is the workflow with `approved` and `received` files ... but now fussy on it, especially at this stage.

**Review by andreaTP — APPROVED** [review]

**Pull request merged**